spec/tunnel_spec.rb

```diff
@@ -107,6 +120,106 @@
     end
   end
 
+  context 'test custom configs linux - generated ssh command' do
+    let(:mock_generated_cmd) do
+      'mock_ssh_cmd_bin -o option1 -o option2 mock_user@mock_host'
+    end
+
+    before do
+      TasteTester::Config.ssh_connect_timeout 10
+      TasteTester::Config.jumps 'mock_jump_user@mock_jump_host'
+      TasteTester::Config.user 'rossi'
+      TasteTester::Config.ssh_cmd_gen_template 'mock_generator_cmd ' +
+        'mock_arg1 mock_arg2 %{jumps} %{host} --user %{user} --get-command'
+      TasteTester::Config.use_ssh_tunnels true
+      allow(mock_so).to receive(:run_command).and_return(mock_so)
+      allow(mock_so).to receive(:error?).and_return(false)
+      allow(mock_so).to receive(:error!).and_return(mock_generated_cmd)
+      allow(mock_so).to receive(:stderr).and_return('')
+      allow(mock_so).to receive(:stdout).and_return(mock_generated_cmd)
+    end
+
+    it 'test ssh generated command' do
```

Comment:
minor nit, the tests (the `it` statements) should be named for what the thing is supposed to DO. So it should be like `it "generates commands"`. Meanwhile the "contexts" should be "when" statements to describe the context. because then the output is like:
```
* when using custom generators
  * it generates a proper non-tunnel command
  * it generates a proper tunnel command
```

Reply:
Oh! lol I guess I should...

[https://c.tenor.com/RUJokzj0C2gAAAAC/speak-english-pulp-fiction.gif](url)